adr/0007-choosing-an-open-source-component.md

```diff
@@ -0,0 +1,57 @@
+# Choosing an open source component
+
+Date: 2018-01-09
+
+## Status
+
+Proposed.
+
+## Context 
+
+Using poor quality or unmaintained third-party software can be at the least inconvenient, and at the most downright dangerous: for example, it can fail to take advantage of the benefits from the latest technology, and security holes can go unpatched.
+
+Using third-party software that is difficult to maintain integrated in our environment can make updating it difficult, time-consuming or risky.
+
+Using software with an inappropriate license may present a legal risk, or may risk others choosing not to use our software because of the burden the license subsequently places on them.  
+
+We want to minimise exposure to these risks when choosing an open source component to use in the eLife code base.   
+            
+## Decision
+When choosing an open source component, care should be taken to try to choose something that is:
+
+- licensed appropriately
+- reliable
+- well maintained
+- sustainable within our environment
+
+### Licensing
+
+[should we specify this in terms of an explicit list of acceptable licenses, or in terms of what we need a licence to allow us to do?]
+
+### Reliability
+
+- Check the code: are there any obvious code smells?
+- Check the documentation: is there a README with a quickstart guide to get you up and running quickly? Collectively, is the documentation sufficient?
+- Try out the code: does it behave as expected? Are there any idiosyncrasies in behaviour or usage that may indicate a design problem? A design problem may make bugs more likely, and may make it harder to fix them.  
+- Check the issues (open and closed): how many bug reports are there? How frequently are they reported?
+
+These steps should help form a view of the likely reliability of the component.
+
+### Well maintained
+
+- Are there a decent number of recent commits? This will indicate whether it's under active development. Avoid it if it's not.
```

Comment:
I see this stance a lot in javascript developers. recent activity is no indication of stability and is no reason to avoid a thing.

Reply:
I think this is spuriously correlated with how big a project is, e.g. the Linux kernel has many commits per day even if some areas are very stable and the release pipeline is very long.

Reply:
In the other direction, if there is a single master branch with many commits per week it's an indication of _in_-stability. Number of existing releases could be a better litmus test for stability?

Reply:
In the JavaScript world things are moving so fast that something which lacks recent commits probably is out of date. I take the point that this doesn't necessarily apply in other environments.